kind/pkg/cluster/kubeadm/config.go

```diff
@@ -31,19 +31,19 @@ type ConfigData struct {
 	KubernetesVersion string
 	// UnifiedControlPlaneImage - optional
 	UnifiedControlPlaneImage string
-	// AutoDerivedConfigData is populated by DeriveFields()
-	AutoDerivedConfigData
+	// DerivedConfigData is populated by DeriveFields()
+	DerivedConfigData
 }
 
-// AutoDerivedConfigData fields are automatically derived by
-// ConfigData.DeriveFieldsif they are not specified / zero valued
-type AutoDerivedConfigData struct {
+// DerivedConfigData fields are automatically derived by
+// ConfigData.Derive if they are not specified / zero valued
+type DerivedConfigData struct {
 	// DockerStableTag is automatically derived from KubernetesVersion
 	DockerStableTag string
 }
 
-// DeriveFields automatically derives DockerStableTag if not specified
-func (c *ConfigData) DeriveFields() {
+// Derive automatically derives DockerStableTag if not specified
+func (c *ConfigData) Derive() {
```

Comment:
Would `Derive` be better called `Default`? This seems like another form of configuration to me 😄

Reply:
I didn't want to call this `default` as these fields are not meant to ever be manually set, you should normally specify the other fields and then when this struct is used .Derive() will be called first to populate these fields.